app/js/features/escrow/saga.js

```diff
@@ -0,0 +1,27 @@
+/*global web3*/
+import Escrow from 'Embark/contracts/Escrow';
+
+import { fork, takeEvery, call, put } from 'redux-saga/effects';
+import {CREATE_ESCROW, CREATE_ESCROW_FAILED, CREATE_ESCROW_SUCCEEDED} from './constants';
+
+const zeroAddress = '0x0000000000000000000000000000000000000000';
+
+export function *createEscrow({expiration, value, buyer}) {
+  try {
+    // TODO do we want to change the token or always ETH?
+    expiration /= 1000; // Solidity is in seconds, while this is milliseconds
+    const toSend = Escrow.methods.create(buyer, parseInt(value, 10), zeroAddress, expiration);
+    const estimatedGas = yield call(toSend.estimateGas, {value});
+    const receipt = yield call(toSend.send, {gasLimit: estimatedGas + 1000, from: web3.eth.defaultAccount, value});
+    yield put({type: CREATE_ESCROW_SUCCEEDED, result: receipt});
```

Comment:
This is really nice but I would not rename it and keep it receipt, even in the state, that way I know immediately that a result is a receipt in other layer

Reply:
Makes sense